### ISSUE TITLE
[Feature Request] Delete SubscriptionItem should take options to set `prorate = false`

### ISSUE THREAD

**Issue by JessePerry** [stripe/stripe-dotnet#1931]
This is a feature request or a missing feature.
Create and Update SubscriptionItem can be called with options like this:
```
await subscriptionItemService.UpdateAsync(subItem.Id, new SubscriptionItemUpdateOptions { Quantity = requestedQuantity, Prorate = _isProrate });
```
But Delete has no such options object or method. So if you delete a sub item with this api client, it will always prorate=true.

In the api docs, it's possible to give options to [delete ](https://stripe.com/docs/api/subscription_items/delete) 

Delete and DeleteAsync should take a parameter SubscriptionItemDeleteOptions to mimic the ability of Create/update and allow deletion with prorata = false.

**Comment by remi-stripe** [comment]
@JessePerry Thanks a lot for raising this! Today, the majority of the Delete operations don't support passing any parameters and this endpoint is more an exception.

It seems we never fixed the library to properly reflect that. I'm going to work on a fix which will be straightforward but it will be a breaking change so it will be released in the next major version that is coming out next week.

**Comment by remi-stripe** [comment]
Okay PR in progress here https://github.com/stripe/stripe-dotnet/pull/1932 if you want to follow along!

**Comment by ob-stripe** [comment]
This was fixed in v35.0.0.
